contracts/MyNFT.sol

```diff
@@ -12,20 +12,27 @@ contract MyNFT is ERC721URIStorage, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter private _tokenIds;
 
+    Counters.Counter private randNonce;
+
+    uint256 private lastRaceResult = 0;
+
     constructor() ERC721("MyNFT", "NFT") {}
 
-    function random() private view returns (uint256) {
+    function random() private returns (uint256) {
         // sha3 and now have been deprecated
-        return
-            uint256(
-                keccak256(
-                    abi.encodePacked(
-                        block.difficulty,
-                        block.timestamp,
-                        _tokenIds.current()
-                    )
+        randNonce.increment();
```

Comment:
without this we were getting same result each time.

Reply:
cool, still a bad random prone to attacks but cool that it works :)